### DISCUSSION TITLE
python setup.py install error

### DISCUSSION THREAD

**Discussion by nsano-rururu** [jertel/elastalert2#884] [Ideas]
elastalert 2.5.1

```
python setup.py install
```

```sh
Installed /home/sano/py310/lib/python3.10/site-packages/pytz-2022.1-py3.10.egg
error: requests 2.28.0 is installed but requests==2.27.1 is required by {'tencentcloud-sdk-python'}
```
**fix setup.py**

```
'requests>=2.27.1',
```
to
```
'requests==2.27.1',
```

**fix requirements.txt**

requirements.txt
```
requests>=2.27.1
```
to
```
requests==2.27.1
```

**Comment by jertel** [comment]
On Jan 16, a PR was merged into the `tencentcloud-sdk-python` project that added this dependency. The original commit did not specify a `requests` lib version, but a code review comment requested the version be included.

https://github.com/TencentCloud/tencentcloud-sdk-python/pull/121#discussion_r782076828

It would help our project and likely many others out there if they did not pin to that specific version, and instead allowed for a range or minimum version. If we choose to pin ElastAlert2 version to that specific requests version it will then likely cause problems for those who do not use the ElastAlert2 docker, and have other Python apps running in that same environment.

An alternative option is to rollback the `tencentcloud-sdk-python` to a version released before that PR was merged.

We can also request that the `tencentcloud-sdk-python` project unpin to that specific version.

**Comment by jertel** [comment]
Let's see what they say. 

https://github.com/TencentCloud/tencentcloud-sdk-python/issues/133

**Reply by jertel** [reply]
They have closed the issue with acknowledgement that pinning it to the minor version was a mistake. It will be corrected in the next release. They do releases every couple of days, so the install issue should be resolved soon.
